src/controllers/sign_s3.js

```diff
@@ -0,0 +1,32 @@
+const aws = require('aws-sdk');
+require('env2')('./config.env');
+const S3_BUCKET = process.env.S3_BUCKET;
+
+// set the region of the S3 bucket
+aws.config.region = 'eu-west-2';
+
+module.exports = (req, res) => {
+  const s3 = new aws.S3();
+  const fileName = req.query['file-name'];
+  const fileType = req.query['file-type'];
+  const s3Params = {
+    Bucket: S3_BUCKET,
+    Key: fileName,
```

Comment:
Do not let the user have control over the s3 key. Also, the domain the user can write to should be limited.

ie
```js
key = `${hashSha256(userId)}:${uuidv4()}`
```
With your current approach, an two files uploaded with the same name would overwrite each other.

Hashing the user id may not be 100% necessary